src/features/authentication/login/login.tsx

```diff
@@ -14,254 +13,233 @@ import { Checkbox } from "primereact/checkbox";
-      try {
-        await signInWithPopup(auth, provider);
-        localStorage.setItem("authenticated", JSON.stringify(true));
-        navigate("/homepage");
-      } catch (error: any) {
-        console.log(error.message);
-      }
-    };
-
-    const signInWithFacebook = async () => {
-      const provider = new FacebookAuthProvider();
-      try {
-        await signInWithPopup(auth, provider);
-        localStorage.setItem("authenticated", JSON.stringify(true));
-        navigate("/homepage");
-      } catch (error: any) {
-        console.log(error.message);
-      }
-    };
-
-    const handleChange = (e: any) => {
-      const { name, value } = e.target;
-      setFormValues((prevState) => ({ ...prevState, [name]: value }));
-    };
-
-    const handleSubmit = (e: any) => {
-      e.preventDefault();
-      setFormErrors(getErrors(formValues));
-      setIsSubmitPress(true);
-    };
-
-    const getErrors = (values: any) => {
-      const validationErrors = initialValues;
-
-      const regex = /^[\w-\.]+@([\w-]+\.)+[\w-]{2,4}$/i;
-      if (!values.email) {
-        validationErrors.email = "Email is required.";
-      } else if (!regex.test(values.email)) {
-        validationErrors.email = "This is not a valid email format!";
-      }
-      if (!values.password) {
-        validationErrors.password = "Password is required.";
-      } else if (values.password.length < 8) {
-        validationErrors.password = "Password must be more than 8 characters.";
-      }
-      return validationErrors;
-    };
-
-    useEffect(() => {
-      // validating form errors to see if we can loggin to app
-      if (
-        !formErrors.email.length &&
-        !formErrors.password.length &&
-        isSubmitPress
-      ) {
-        loginToApp();
-      }
-    }, [formErrors, isSubmitPress]);
-
-    useEffect(() => {
-      onAuthStateChanged(auth, (currentUser) => {
-        let user;
-        if (currentUser) {
-          currentUser.getIdTokenResult().then((idTokenResult) => {
-            const isAdmin = !!idTokenResult.claims.admin;
-            user = { ...currentUser, isAdmin };
-            localStorage.setItem("user", JSON.stringify(user));
-            setUser(user);
-          });
-        }
-      });
-    }, []);
-
-    return (
-      <div className="login">
-        <div className=" align-items-center justify-content-center card--center">
-          <div className="login__card">
-            <div className="text-center mb-2">
-              <img src={logo} alt="hyper" height={50} className="login__logo" />
-            </div>
+    }
+  };
+
+  const signInWithGoogle = async () => {
+    const provider = new GoogleAuthProvider();
+    try {
+      await signInWithPopup(auth, provider);
+      localStorage.setItem("authenticated", JSON.stringify(true));
+      navigate("/homepage");
+    } catch (error: any) {
+      console.log(error.message);
+    }
+  };
+
+  const signInWithFacebook = async () => {
```

Comment:
The functions signInWithGoogle and signInWithFacebook do the same thing, so it would be cleaner to have a single function which takes in the provider as the parameter and does the login based on that.